### ISSUE TITLE
Orphaned-Texts not working on the Google Nexus 4 (16GB) with pure Google Android 4.4.2

### ISSUE THREAD

**Issue by TheKisho** [michalmazur/orphaned-texts#11]
Orphaned-Texts use to work for me on my Nexus 4 pre-December 2013.  I think a combination of the November 27 (v1.2) update and the Android update to 4.4 (and minor versions) [Never received the 4.3.x updates, skipped from 4.2.x to 4.4.x] may of killed it on my Google Nexus 4 (from Google, not carrier's phone).  Both updates happened around that time and quickly so I was unable to judge.    v1.1 and Android 4.2.x worked perfectly.  The application runs and collects orphaned text messages, I can clear them out but they keep coming back (earliest one is December 6th).  The application seems to keep acquiring more, and it no longer seems to stop text message corruption.  Text messaging is starting to become unusable, I hope you can fix this application because I desperately need it.  v1.2 and v1.3 on Android 4.4.2 hasn't been working for me. :S  If you require any further information which can help resolve this issue, I would be glad to provide in hopes that is can be resolved.

**Comment by AnAkkk** [comment]
Same, I have Android 4.2.2 (CyanogenMod 11) on my I9100G, and Orphaned-Texts doesn't work either.

**Comment by michalmazur** [comment]
Thank you for reporting the problems.

@TheKisho I have not made any significant changes to the way this app works since the initial release on Play Store. How soon do the messages reappear? The app does not check how many parts are actually removed; perhaps this is something that I could improve in a future release.

@AnAkIn1 Do you mean that removed message parts also reappear on your device?

In the meantime you may try Ian's similar app https://code.google.com/p/android/issues/detail?id=28697#c5

**Comment by TheKisho** [comment]
Soon as the phone is restarted, or the app is removed from recent apps list (never was an issue in the past).  (Not sure how long it takes on it's own before the texts reappear.  I've notice they will if I do nothing and let it idle in the background).  I use the Messaging app still (default Android; I don't like using Hangouts for text messages).  As I said I cannot pin point if it was the newer versions of Android or your program that caused the issue.  (Perhaps a new kernel for this Nexus 4 (wasn't paying to close to the deeply nerdy stuff Google updated or changed version to version)).  I'm thinking it is just an interaction issue, perhaps something Google broke.  I don't know much about this, I just know it isn't behaving properly (and isn't working).  Seems like it isn't committing the wipes of the orphan messages permanently, as well as not stopping the messages (not sure if it actually doing anything when it show a blank list once deleted).  :S

Yeah I was using that before you released this program.  Not sure if it is working properly anymore, I'll give it a try in a bit.

**Comment by AnAkkk** [comment]
Yes, that's what I meant. I was using the app you linked, and had the exact same issue, that's why I tried your app.

**Comment by KeloneW** [comment]
Same problem on my sister's phone (Galaxy SIII with android 4.2.2), messages keeps reappearing when she re-open the app.

**Comment by TheKisho** [comment]
As of the March 1, 2014 update it now reports "0 messages were deleted" when deleting all, and no messages go away.  The overall problem this program is suppose to solve is still occurring.  I really would enjoy this app to work once again, it is a highly useful application I recommend to all my friends using android.  I would pay for if it wasn't free, it is that valuable.  There is nothing like it, the above application which I also use to use before this appeared in the Google Play store, doesn't function for me anymore, also.    Another person in the Google Play store user reviews reports that it [OT] doesn't work with their Android 4.4.2 phone as well (previous version of OT).  I'm unsure why the application is unable to delete these message fragments, but it is still highly likely Android has changed something behind the scenes if you haven't changed anything between versions.

**Comment by michalmazur** [comment]
While I do not have access to any devices running Android 4.4 I tested the app in the emulator and unfortunately as of 4.4 it is no longer possible to add or delete message parts. The least I could do in Orphaned Texts 1.4 was to stop erroneously reporting successful removal of orphaned message parts. I will try to pull the app from Play Store for devices running Android 4.4 and above.

Orphaned Texts behaves correctly in the Android 4.2.2 emulator. Unfortunately, I have no idea why it is not working correctly on Samsung Galaxy phones running Android 4.2.2.

**Comment by HughNoble** [comment]
Is there another way to remove these messages on 4.4? It's annoying as I have orphaned messages that I can not remove.

**Comment by michalmazur** [comment]
Not that I know of, unfortunately.

**Comment by michalmazur** [comment]
@TheKisho, @HughNoble Just an update: I released a beta version of [Orphaned Texts for KitKat](https://play.google.com/store/apps/details?id=com.michalmazur.orphanedtexts.kitkat) on Play Store. Please give it a try if orphaned messages are still a problem for you.
